controlplane/rosa/controllers/rosacontrolplane_controller_test.go

```diff
@@ -203,6 +203,34 @@ func TestUpdateOCMClusterSpec(t *testing.T) {
 		g.Expect(updated).To(BeTrue())
 		g.Expect(ocmSpec).To(Equal(expectedOCMSpec))
 	})
+
+	// Test case 7: AutoNode update
+	t.Run("Update Auto Node", func(t *testing.T) {
+		rosaControlPlane := &rosacontrolplanev1.ROSAControlPlane{
+			Spec: rosacontrolplanev1.RosaControlPlaneSpec{
+				AutoNode: &rosacontrolplanev1.AutoNode{
+					Mode:    rosacontrolplanev1.Enabled,
+					RoleARN: "autoNodeARN",
+				},
+			},
+		}
+
+		mockCluster, _ := v1.NewCluster().
+			AutoNode(v1.NewClusterAutoNode().Mode("disabled")).
+			AWS(v1.NewAWS().AutoNode(v1.NewAwsAutoNode().RoleArn("anyARN"))).
+			Build()
+
+		expectedOCMSpec := ocm.Spec{
+			AutoNodeMode:    "enabled",
+			AutoNodeRoleARN: "autoNodeARN",
+		}
+
+		reconciler := &ROSAControlPlaneReconciler{}
+		ocmSpec, updated := reconciler.updateOCMClusterSpec(rosaControlPlane, mockCluster)
+
+		g.Expect(updated).To(BeTrue())
+		g.Expect(ocmSpec).To(Equal(expectedOCMSpec))
+	})
```

Comment:
Once we have polished the API above let's define unit tests for the various cases where things are left empty.